src/main/java/tds/content/controllers/ContentController.java

```diff
@@ -108,11 +112,15 @@ public ContentController(final ContentService contentService) {
     @ResponseBody
     public ResponseEntity<String> getItemData(@RequestParam final String itemPath) throws IOException {
         try {
-            return ResponseEntity.ok(contentService.loadData(new URI(itemPath)));
+            return ResponseEntity.ok(contentService.loadData(getEncodedUri(itemPath)));
         } catch (URISyntaxException e) {
             throw new IllegalArgumentException(String.format("The provided item path '%s' was malformed", itemPath));
         } catch (final FileNotFoundException ex) {
             return new ResponseEntity<>(ex.getMessage(), HttpStatus.NOT_FOUND);
         }
     }
+
+    private URI getEncodedUri(final String uriPath) throws URISyntaxException, UnsupportedEncodingException {
```

Comment:
Same theme but different question, should this be throwing 500?  It seems like this should catch exceptions and maybe throw an IllegalArgumentException which then will result in a 400 Http response code since the data passed in is junk.